app/views/layouts/application.html.haml

```diff
@@ -6,7 +6,7 @@
     - if logged_in?
       %nav.admin_panel
         %ul
-          %li= link_to current_developer.username, current_developer
+          %li= link_to current_developer.username, developer_path(current_developer.username)
```

Comment:
did you try overriding the method `to_param` in developer model? I am pretty sure it fixes it as well, just not sure which fix would be better, or maybe both

Reply:
That's an interesting approach! I'll take a look

Reply:
Totally a better approach, changing it! Thanks @softr8